b2sdk/sync/sync.py

```diff
@@ -264,12 +259,12 @@ def make_folder_sync_actions(
             elif dest_file is None:
                 logger.debug('determined that %s is not present on destination', source_file)
 
-            if source_folder.folder_type() == 'local':
-                if source_file is not None:
-                    reporter.update_compare(1)
-            else:
-                if dest_file is not None:
-                    reporter.update_compare(1)
+            if source_file is not None:
+                if source_type == 'b2':
+                    # For buckets we don't want to count files separately as it would require
+                    # more API calls. Instead, we count them when comparing.
+                    reporter.update_total(1)
```

Comment:
won't that number fall short if `delete=True` and the destination has way more files than the source?

Reply:
We must decide from where we're counting files (from src or dst). The previous implementation was really odd (always counted on local). It doesn't make any sense.

The other way is to completely remove counting from the report.

Reply:
Shouldn't those counters be always updated when comparing, getting a total number and not just a number from one side of the sync?

Can we move the updating to always happen during compare?

Reply:
But we have three counters: `total, compared, updated`. The first two could be merged, but the `total` counter is actually useful when you sync large amounts of files from local. When the source is a bucket (either for copy or download) we can not calculate total files faster than during comparison. At least in the current implementation, without significant change.

Reply:
Ok, please file a ticket for a possible improvement in this space, it would make progress reporting better

Reply:
I will